test/common/upstream/load_balancer_fuzz_base.cc

```diff
@@ -0,0 +1,193 @@
+  const uint8_t priority_of_host_set = host_priority % num_priority_levels_;
+  ENVOY_LOG_MISC(trace, "Updating health flags for host set at priority: {}", priority_of_host_set);
+  MockHostSet& host_set = *priority_set_.getMockHostSet(priority_of_host_set);
+  // This downcast will not overflow because size is capped by port numbers
+  const uint32_t host_set_size = host_set.hosts_.size();
+  host_set.healthy_hosts_.clear();
+  host_set.degraded_hosts_.clear();
+  host_set.excluded_hosts_.clear();
+
+  Fuzz::ProperSubsetSelector subset_selector(random_bytestring);
+
+  const std::vector<std::vector<uint8_t>> subsets = subset_selector.constructSubsets(
+      {num_healthy_hosts, num_degraded_hosts, num_excluded_hosts}, host_set_size);
+
+  // Healthy hosts are first subset
+  for (uint8_t index : subsets.at(0)) {
+    host_set.healthy_hosts_.push_back(host_set.hosts_[index]);
+  }
+  ENVOY_LOG_MISC(trace, "Hosts made healthy at priority level {}: {}", priority_of_host_set,
+                 absl::StrJoin(subsets.at(0), " "));
+
+  // Degraded hosts are second subset
+  for (uint8_t index : subsets.at(1)) {
+    host_set.degraded_hosts_.push_back(host_set.hosts_[index]);
+  }
+  ENVOY_LOG_MISC(trace, "Hosts made degraded at priority level {}: {}", priority_of_host_set,
+                 absl::StrJoin(subsets.at(1), " "));
+
+  // Excluded hosts are third subset
+  for (uint8_t index : subsets.at(2)) {
+    host_set.excluded_hosts_.push_back(host_set.hosts_[index]);
+  }
+  ENVOY_LOG_MISC(trace, "Hosts made excluded at priority level {}: {}", priority_of_host_set,
+                 absl::StrJoin(subsets.at(2), " "));
+
+  // Handle updating health flags for hosts_per_locality_
+
+  // The index within the array of the vector represents the locality
+  std::array<HostVector, 3> healthy_hosts_per_locality;
+  std::array<HostVector, 3> degraded_hosts_per_locality;
+  std::array<HostVector, 3> excluded_hosts_per_locality;
+
+  // Wrap those three in an array here, where the index represents health flag of
+  // healthy/degraded/excluded, used for indexing during iteration through subsets
+  std::array<std::array<HostVector, 3>, 3> locality_health_flags = {
+      healthy_hosts_per_locality, degraded_hosts_per_locality, excluded_hosts_per_locality};
+
+  // Iterate through subsets
+  for (uint8_t health_flag = 0; health_flag < locality_health_flags.size(); health_flag++) {
+    for (uint8_t index : subsets.at(health_flag)) { // Each subset logically represents a health
+                                                    // flag
+      // If the host is in a locality, we have to update the corresponding health flag host vector
+      if (!(locality_indexes_.find(index) == locality_indexes_.end())) {
+        // First dimension of array represents health_flag, second represents locality, which is
+        // pulled from map
+        locality_health_flags[health_flag][locality_indexes_[index]].push_back(
```

Comment:
I think it would be clearer to do the `locality_index_` find above and bind to a local variable (with a meaningful name).

Reply:
Changed.